frame/staking/src/lib.rs

```diff
@@ -1939,12 +2148,18 @@ impl<T: Trait> Module<T> {
 	///
 	/// This data is used to efficiently evaluate election results. returns `true` if the operation
 	/// is successful.
-	fn create_stakers_snapshot() -> bool {
+	fn create_stakers_snapshot() -> (bool, Weight) {
```

Comment:
I would add this to a list of _potentially very large weights_ that we should check with some realistic values from a normal chain (1000 validators, 200,000 nominators) and see what is the outcome. I don't think we can do much about them but still it would be good to know if we have weight formulas that could potentially grow very large.